daras_ai_v2/base.py

```diff
@@ -2425,6 +2450,42 @@ def is_current_user_paying(self) -> bool:
     def is_current_user_owner(self) -> bool:
         return bool(self.request.user and self.current_sr_user == self.request.user)
 
+    def is_user_authorized(self, user: AppUser | None = None) -> bool:
+        if self.is_user_admin(user):
+            return True
+
+        sr, pr = self.current_sr_pr
+        if pr.saved_run_id != sr.id:
+            # not published run: simply check if user is a member of workspace
+            return bool(user) and sr.workspace_id in user.cached_memberships
+
+        # published run
+        if pr.is_approved_example:
+            return True
+
+        if WorkflowAccessLevel(pr.public_access) in (
+            WorkflowAccessLevel.FIND_AND_VIEW,
+            WorkflowAccessLevel.EDIT,
+        ):
+            # public access allowed
+            return True
+
+        membership = user and user.cached_memberships.get(pr.workspace_id, None)
+        role = membership and WorkspaceRole(membership.role)
+        match pr.workspace_access, role:
```

Comment:
i believe all members should be able to see any run including WorkflowAccessLevel.VIEW_ONLY ones